### PR TITLE
Updated Authenticator to v5.0.0

### PR DESCRIPTION
**Pull request by akopachov**
# Application Submission

Updated Authenticator app to [v5.0.0](https://github.com/akopachov/flipper-zero_authenticator/releases/tag/v5.0.0)

# Extra Requirements 

Authenticator CLI was extracted into separate FAL plugins which are getting loaded on demand. So please make sure that your bundles and installers do care about putting not just FAP but also all the FAL files when app is installing to the device

# Author Checklist (Fill this out)

- [x] I've read the [contribution guidelines](../blob/HEAD/documentation/Contributing.md) and my PR follows them
- [x] I own the code I'm submitting or have code owner's permission to submit it
- [x] I [have validated](../blob/HEAD/documentation/Contributing.md#validating-manifest) the manifest file(s)

# Reviewer Checklist (Don't fill this out)

- [ ] Bundle is valid
- [ ] There are no obvious issues with the source code
- [ ] I've ran this application and verified its functionality

### CODE REVIEW

**Comment by hedger** [comment]
Unfortunately, at the moment .fals are not supported by app delivery system.

**Comment by akopachov** [comment]
Is it planned to implement?

**Comment by hedger** [comment]
Yes, in long-term plans.

**Comment by akopachov** [comment]
Okay, how would you like this PR to be handle? 
- Leave it here open until FAL support arrives?
- Close PR, open issue and open new PR when FAL support arrives?
- Any other way?

**Comment by hedger** [comment]
We'll come back after discussing that later today.

**Comment by hedger** [comment]
OK - so we decided to add an option to the build system to allow packaging .fal files as resources, in order to keep current  app distribution scheme in form of a single file.

That will require some changes on apps' side - .fals that are packaged that way will be extracted to `apps_assets`.

Build system changes are in the works.

**Comment by akopachov** [comment]
So just to make sure I understand correctly:
- OFW catalog build & deploy pipeline will pack FAL files into FAP and then they will be extracted into `apps_assets/plugins/`
- `./fbt launch APPSRC=totp` still will be deploying FAL files independently to `apps_data/totp/plugins/`?

**Comment by hedger** [comment]
Nope. Plugins will be getting a new `fal_embedded` manifest option, which will make the build system embed that plugin into assets of parent app(s). It will work that way for all methods of building and launching the application.

**Comment by hedger** [comment]
Please check out the initial implementation of the feature in https://github.com/flipperdevices/flipperzero-firmware/pull/3083

tldr: mark your plugins with `fal_embedded=True`

**Comment by akopachov** [comment]
Thanks for the implementation.
Is there any way to identify at compile time (via `#ifdef` or so) whether `fal_embedded` is enabled\supported or not? Just want to update the code in such a way that it will try to load plugins from `apps_assets` if `fal_embedded` was enabled and supported by build pipeline, or from `apps_data` if `fal_embedded` was turned off or not supported by build pipeline.

**Comment by hedger** [comment]
Nope, we are expecting all firmware versions to maintain same set of basic build system capabilities.

**Comment by hedger** [comment]
I understand that you have already developed an advanced scripting system around full firmware sources for building your app - but, I suggest you switch to ufbt and using SDKs for different firmwares. That would reduce complexity of your builds and make them easier to maintain.

If you need any extra features from the build system, feel free to request them.

**Comment by akopachov** [comment]
I was thinking about switching to uFBT, but what was really stopped me is that it was pretty unclear from uFBT docs how to:
1. Use uFBT to build an app using custom firmware (let's be honest - there is custom firmwares, app is compatible with their extended API, so I'd like to be able to have a builds specifically for that firmwares too)
2. How to do global defines (something like `-DFEATURE` or `-DFEATURE=2`)

So I just postponed it for some time. If you have some good examples or good explanations how to achieve two goals above - I'm all in to move to uFBT and actually simplify my custom build scripts

**Comment by hedger** [comment]
1. uFBT can download SDKs from various sources - direct links or using indexers compatible with the one official firmware uses, like Unleashed does. Check out [this section](https://github.com/flipperdevices/flipperzero-ufbt#managing-the-sdk) for details.

2. Some time ago, we introduced global defines for firmwares that include their identity - `FW_ORIGIN_*`. For example, official firmware defines `FW_ORIGIN_Official`, Unleashed has `FW_ORIGIN_Unleashed`, etc. That feature was intended for replacing user-defined feature flags. Hope that satisfies your needs.

**Comment by akopachov** [comment]
Yeah, it does help. Will try again with uFBT and will definitely come back if any questions. Thanks

**Comment by akopachov** [comment]
May I request a few help\clarification [here](https://github.com/akopachov/flipper-zero_authenticator/tree/fal_embedded_incorp)?

* I migrated all the build stuff to uFBT (and it is working pretty good)
* I added `fal_embedded=True` to all the plugins
* When trying to build&launch an app using uFBT it fails with weird issue `scons: *** [C:\Users\alexk\.ufbt\build\totp.fap] TypeError : Tried to lookup Dir 'C:\Workspace\Projects\flipper-zero_authenticator\totp\assets\cli\cli' as a File.`

Attaching [buildlog.txt](https://github.com/flipperdevices/flipper-application-catalog/files/12713902/buildlog.txt)
